packages/web-app-preview/src/App.vue

```diff
@@ -233,15 +161,47 @@ export default defineComponent({
       }
     }
 
+    const filteredFiles = computed<Resource[]>(() => {
+      if (!unref(activeFiles)) {
+        return []
+      }
+
+      const files = unref(activeFiles).filter((file) => {
+        return mimeTypes().includes(file.mimeType?.toLowerCase())
+      })
+
+      return sortHelper(files, [{ name: unref(sortBy) }], unref(sortBy), unref(sortDir))
+    })
+    const activeFilteredFile = computed(() => {
+      return unref(filteredFiles)[unref(activeIndex)]
+    })
+    const activeMediaFileCached = computed(() => {
+      return unref(cachedFiles).find((i) => i.id === unref(activeFilteredFile).id)
+    })
+
+    const updateLocalHistory = () => {
+      const { params, query } = createFileRouteOptions(
+        unref(unref(currentFileContext).space),
+        unref(activeFilteredFile)
+      )
+      router.replace({
+        ...unref(route),
+        params: { ...unref(route).params, ...params },
+        query: { ...unref(route).query, ...query }
+      })
+    }
+
```

Comment:
I really like where this PR is going! 

It doesn't need to be implemented in this PR and it probably makes sense to do it, when we introduce another app that needs it, but long term I would really like to separate this logic from the app.

Reply:
I'm not 100% sure where your comment belongs to, it kinda sits at the end of `updateLocalHistory` - you mean that method in general? 😄

Reply:
Nah, all of that file switching, route, filtering logic. 

I would like to implement just the filter in this app and have eVerything else handled by a composable